velox/common/memory/HashStringAllocator.cpp

```diff
@@ -90,7 +90,10 @@ void HashStringAllocator::clear() {
   freeBytes_ = 0;
   std::fill(std::begin(freeNonEmpty_), std::end(freeNonEmpty_), 0);
   for (auto& pair : allocationsFromPool_) {
-    pool()->free(pair.first, pair.second);
+    auto size = pair.second;
+    pool()->free(pair.first, size);
+    sizeFromPool_ -= size;
+    cumulativeBytes_ -= size;
```

Comment:
Can we add a check in ~HashStringAllocator to verify both sizeFromPool_ and cumulativeBytes_ are zero to help us to catch this sort of problem in production? The downside is that we might see crash but we should fix anyway? Thanks!